### ISSUE TITLE
Delay for tooltip

### ISSUE THREAD

**Issue by anatfenn** [wix/wix-style-react#2867]
# ✨ Feature Request

### 🔦 Scope

...

### 💁 Explanation

...

### 🐾 Possible solution 

...

### 👀 Severity

- Critical / Major / Low

**Comment by anatfenn** [comment]
allow adding a delay for tooltip appearance

![ea540d06d0d9db86c4b1384dd9a81c74](https://user-images.githubusercontent.com/43814181/52064052-77f18480-257c-11e9-9f9d-567e543d22f9.gif)

**Comment by shlomitc** [comment]
@wuwa @milkyfruit 
We can add the configuration, however, I think the initial show delay should be defined as built-in behavior of the component.

I think that the native is 3 seconds +-. What do you think?

**Comment by milkyfruit** [comment]
Yes, I agree it should be a toggle between instant/delayed states. We shouldn't allow configuring the timing. Later that transition should become part of futures animation/transition library.

The exact delay values @zilvinas will share later today.

**Comment by ZilvinasMaciulis** [comment]
For now, let's stick with:
On Mouse In: Delay 0.35
On Mouse Out: Delay 0

In the future, when we will define animations it should be: 
On Mouse In: Element feedback(100ms)>Delay(XX)>Tooltip Animate In (XX)
On Mouse Out: Element feedback(100ms)+Tooltip Animate Out(XX)

**Comment by shlomitc** [comment]
1. Is this for all tooltips that opens on hover or just for the ellipsis tooltip?
2. What’s the difference between feedback, delay and animate? Can you describe it a bit more visually

**Comment by milkyfruit** [comment]
@shlomitc 
1. Right now, for ellipsis only.
2. Here are definitions:
- Element Feedback – usually, it's mouse hover effect of an element that triggers a tootlip
For example: iconButton
- Delay – it's a simple delay of Animate in/out after it was triggered. 
For example: `animation-delay: 2s;`
- Animate In – it's an animated visual transition making an element appear on a screen
For example:`
@keyframes fadeIn {
  from {opacity: 0}
  to {opacity: 1}
}
`

**Comment by ipanasenko** [comment]
Hey, I want to help with implementing this delay, but I'm a little confused with all this stuff coming from `wix-ui-core`. Anyone willing to help me, or you willing to better do it yourserlves?

**Comment by mykas** [comment]
@shlomitc

**Comment by shlomitc** [comment]
@ipanasenko - yes, sorry about the complecation, it does require some changes in `wix-ui-core` and `wix-ui-backoffice` to allow the Tooltip animation to be configured. we'll try to make it simple in the future but currently it's mandatory

### wix-ui-core
1. `withEllipsedTooltip` - add `tooltipProps` prop to the component, so it can be configured
https://github.com/wix/wix-ui/blob/master/packages/wix-ui-core/src/hocs/EllipsedTooltip/withEllipsedTooltip.tsx#L95

2. `Tooltip` - add `showDelay` and `hideDelay` props, [same as `Popover`](https://github.com/wix/wix-ui/blob/master/packages/wix-ui-core/src/components/popover/Popover.tsx#L70-L73) in order to supply full control over the animations 
https://github.com/wix/wix-ui/blob/master/packages/wix-ui-core/src/components/tooltip/Tooltip.tsx

### wix-ui-backoffice
https://github.com/wix/wix-ui-backoffice/blob/master/src/components/Tooltip/Tooltip.st.css

### wix-style-react
3. In the `wix-style-react` `TextProxy` component, configure the `{tooltipProps : showDelay, hideDelay}`, and css to convey to the UX definitions
